### DISCUSSION TITLE
issue: when i am submitting form it is re rendering whole form component in react native app but that same code is not re rendering in react project

### DISCUSSION THREAD

**Discussion by ritwikchakraborty123** [react-hook-form/react-hook-form#10646] [Q&A] [answered]
### Version Number

7.45.1

### Codesandbox/Expo snack

https://snack.expo.dev/@ritwikchakraborty/form-array

### Expected behaviour

"cnt" variable is number of re render happening on parent form component it should not increase in form submit but not it is increasing every time i am clicking form submit

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@ritwikchakraborty123 

Inside your SocialContainer component, `formState.errors` is subscribed therefore it will trigger re-rendering on root app level. Removing `formState.errors` will solve the issue.

You need to use field state from `Controller` component or `useFormState` hook to isolate each field array item and avoid re-rendering.

Working demo: https://snack.expo.dev/BDJ7fEDiC

**Reply by ritwikchakraborty123** [reply]
Thanks a lot for the response! 
@leapful

**Reply by ritwikchakraborty123** [reply]
@leapful  i have one more requirement so i want to show is from valid above all forms seperately.
Can't use formState.errors bcz it will again re render all components any idea how this thing i can achieve?

**Reply by leapful** [reply]
@ritwikchakraborty123 

You can use `useFormState` hook on a isolated error message component.

Reference with example: https://react-hook-form.com/docs/useformstate

**Reply by ritwikchakraborty123** [reply]
It's not working 
https://snack.expo.dev/@ritwikchakraborty/form-array
i tried that but it is only working when all forms are valid it is not working like independent form and their validation

**Reply by leapful** [reply]
@ritwikchakraborty123 

You need to use `errors` object from `useFormState` with the target index to check for each field array item.

Working demo: https://snack.expo.dev/T2M_DzXrK

**Reply by ritwikchakraborty123** [reply]
thanks a lot bro .
you are pro with this package.
Great help man thank you
